feature(Cotisation Management)

lomamech opened the pull request:
- Implement API for Cotisation Management
- Implement test Integration for Cotisation Management
- Implement client side for cotisation Management
- Update bhAccountSelect to allow the filter of account by Classe
- Fix error in Account Management Set the property classe in Registred and
  Updated Account
- Implement test E2E for Cotisation Management

reference #2292

lomamech committed:
feature(Cotisation Management)
- Implement API for Cotisation Management
-

lomamech committed:
feature(Cotisation Management)
- Implement API for Cotisation Management
- Implement test Integration for Cotisation Management

lomamech committed:
test(Cotisation Management)
- Implement test E2E for Cotisation Management
- Fixe the get of account classe in Account Management

jniles commented on client/src/i18n/en/cotisation.json:
```diff
@@ -0,0 +1,18 @@
+{
+  "COTISATION" : {
+    "ADD_COTISATION" : "Add a Cotisation",
```
It looks like cotisation is [membership fee](https://translate.google.com/#auto/en/cotisation) in English.  I would propose calling it an "employment fee" so it doesn't get confused with other types of fees.

jniles commented on client/src/js/components/bhAccountSelect.js:
```diff
@@ -37,6 +38,9 @@ function AccountSelectController(Accounts, AppCache, $timeout, bhConstants, $sco
     $ctrl.TITLE_ACCOUNT_ID = bhConstants.accounts.TITLE;
 
+    // To filter accounts by class of accounts
+    $ctrl.classe = $ctrl.classe || '';
```
The account class should be the same thing as the account type.  Can you call this account type and use that?

lomamech committed:
Resolve conflict with Branch Master

lomamech committed:
(fix) Fix bad comment

lomamech committed:
Sanitaze and fix bad comment

jniles commented on client/src/modules/cotisations/cotisations.routes.js:
```diff
@@ -0,0 +1,42 @@
+    })
+
+    .state('cotisations.edit', {
+      url : '/:id/edit',
```
:+1:

jniles commented on server/controllers/finance/accounts/index.js:
```diff
@@ -134,19 +137,17 @@ function list(req, res, next) {
-  if (!isNaN(locked)) {
-    sql += ` WHERE a.locked = ${locked}`;
-  }
+  filters.equals('locked');
```
Nice!

jniles commented on test/end-to-end/cotisations/cotisations.page.js:
```diff
@@ -0,0 +1,133 @@
+  /**
+   * forbid deletion of used function
+   */
+  errorOnDeleteCotisation(n) {
```
Can you make this refer to a text instead of an index?  It will make the test much easier to modify in the future.

jniles reviewed (changes requested):
@lomamech, a few comments here.

I'm pretty sure account `classe` and account type_id are the same thing - especially after the modifications by @mbayopanda which added in Actif and Passif and so on.  Could you use account type_id instead?

Otherwise, the code looks pretty good!

lomamech committed:
refactor(Cotisation Management)
- Use Membership Fee instead cotisation in English Translation
- Remove unused function by error in grade, function and cotisation
  Management
